infrastructure/pyd/make_object.d

```diff
@@ -214,6 +217,8 @@ PyObject* d_to_python(T) (T t) {
     } else static if (isDelegate!T || isFunctionPointer!T) {
         PydWrappedFunc_Ready!(T)();
         return wrap_d_object(t);
+    } else static if (isSlice!T) {
+        return d_to_python_numpy_ndarray(t.field);
```

Comment:
Does this lose shape information?

Reply:
No, it does not

Reply:
It works for 2D or 3D shapes?

Reply:
I had tested for 2D slice, seemed to work. Will test again and confirm

Reply:
Okay my bad, the location info is lost when we use `field`, I had tested locally with `ndarray` which saves shape info. What behavior do you suggest. `ndarray` is zero copy for 1D and will do copy for >1 dimensions

Reply:
Option 1: `d_to_python_numpy_ndarray(t.field)` gives you a numpy array which is flattened. Call reshape on it to restore. Pros: No allocation. Cons: This works only for slices which have a stride of 1.

Option 2: Call `ndarray`. Pros: Works generally. Cons: Some allocation. 

I think I'll go with option 2. The copy is not a full copy, it's only a copy of the structure.

Reply:
PS: Actually, upon further inspection, ndarray allocated even for a 1D array, and it does a full copy of the data.

Reply:
Done